core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/SnapshotFileBatchTextSink.java

```diff
@@ -75,12 +75,14 @@ protected void addFileProperties(FileSetProperties.Builder propertiesBuilder) {
    * Config for SnapshotFileBatchTextSink.
    */
   public static class SnapshotFileBatchTextSinkConfig extends SnapshotFileSetBatchSinkConfig {
-    @Description("The Delimiter used to combine the Structured Record fields")
+    @Description("The Delimiter used to combine the Structured Record fields, Defaults to tab")
+    @Nullable
+    @Macro
     private String delimiter;
 
-    public SnapshotFileBatchTextSinkConfig(String name, @Nullable String basePath, String delimiter) {
+    public SnapshotFileBatchTextSinkConfig(String name, @Nullable String basePath, @Nullable String delimiter) {
       super(name, basePath, null);
-      this.delimiter = delimiter;
+      this.delimiter = delimiter == null ? "\t" : delimiter;
```

Comment:
its a little cleaner to have a no-args constructor that calls this one with tab as the delimiter. then you don't need to do the null check in initialize()